Assignment4/carlomagnani_21A285/task07.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1fPaqV0DeCMVNWs9vBCFT3f2j9Gb8wuOF
+
+**Task 07: Querying RDF(s)**
+"""
+
+!pip install rdflib 
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**
+
+# TO DO
+# Visualize the results
+# define namespace ns
+ns = Namespace("http://somewhere#")
+
+#RDFLib
+
+for s, p, o in g.triples((None,RDFS.subClassOf,ns.Person)):
+    print (s)
+
+print(g.value(subject=None,predicate=RDFS.subClassOf,object=s))
+    
+
+# Load Plugin
+from rdflib.plugins.sparql import prepareQuery
+
+#SPARQL
+
+q1 = prepareQuery('''
+  SELECT 
+    ?x
+  WHERE { 
+    ?x rdfs:rdfs:subClassOf/subClassOf* ns:Person 
+  }
+  ''',
+  initNs = { "ns": ns}
+)
+
+for r in g.query(q1):
+  print(r)
+
+**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+"""
+
+# TO DO
+# Visualize the results
+
+#SPARQL
+
+q2 = prepareQuery('''
+    SELECT DISTINCT ?x 
+    WHERE
+    {
+      ?x rdf:type/rdfs:subClassOf* ns:Person .
+    }
+  ''',
+  initNs = { "ns": ns}
+)
+
+for r in g.query(q2):
+  print(r)
+
+  #RDFLib
+ns = Namespace("http://somewhere#")
+
+for s, p, o in g.triples((None,RDF.type,ns.Person)):
+  print (s)
+print (g.value(subject=None,predicate=p,object=(g.value(subject=None,predicate=RDFS.subClassOf,object=o))))
```

Comment:
This may work in this case, but it's not correct. You should iterate over the subclasses.